lib/js/components/Pagination/Pagination.vue

```diff
@@ -0,0 +1,369 @@
+<template>
+	<div
+		:class="{
+			pagination: true,
+			'-compact': layout === PAGINATION_LAYOUTS.COMPACT,
+			'-centered': isCentered,
+		}"
+	>
+		<div class="pagination__itemsWrapper">
+			<div class="pagination__items -default">
+				<template v-for="(n, index) in items">
+					<div :key="index" class="pagination__itemWrapper">
+						<a
+							v-if="isPage(n)"
+							class="pagination__item"
+							:class="{ '-selected': page === n }"
+							@click.prevent.stop="changePage(n)"
+						>
+							{{ n }}
+						</a>
+
+						<span v-else class="pagination__ellipsis">&hellip;</span>
+					</div>
+				</template>
+			</div>
+
+			<div class="pagination__items -compact">
+				<icon-button
+					:size="ICON_BUTTON_SIZES.MEDIUM"
+					:color="ICON_BUTTON_COLORS.NEUTRAL"
+					:icon="ICONS.FA_ANGLE_LEFT"
+					:state="page <= 1 ? ICON_BUTTON_STATES.DISABLED : ICON_BUTTON_STATES.DEFAULT"
+					@click.native="changePage(page - 1)"
+				/>
+
+				<div class="pagination__compactItem">
+					<input
+						class="pagination__input"
+						type="number"
+						:value="page"
+						:min="1"
+						:step="1"
+						:max="lastPage"
+						@input="onInputValueChange"
+					/>
+					<span class="pagination__text">z {{ lastPage }}</span>
+				</div>
+
+				<icon-button
+					:size="ICON_BUTTON_SIZES.MEDIUM"
+					:color="ICON_BUTTON_COLORS.NEUTRAL"
+					:icon="ICONS.FA_ANGLE_RIGHT"
+					:state="
+						page >= lastPage ? ICON_BUTTON_STATES.DISABLED : ICON_BUTTON_STATES.DEFAULT
+					"
+					@click.native="changePage(page + 1)"
+				/>
+			</div>
+		</div>
+
+		<div class="pagination__accessorySlot">
+			<slot name="accessory" />
+		</div>
+	</div>
+</template>
+
+<style scoped lang="scss">
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/typography';
+@import '../../../styles/settings/media-queries';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/shadows';
+@import '../../../styles/settings/radiuses';
+@import '../../../styles/mixins/layout';
+
+$pagination-item-width: 32px;
+$pagination-item-height: 32px;
+
+.pagination {
+	$self: &;
+
+	@include centeredSpread();
+
+	flex-grow: 1;
+	flex-direction: row;
+	align-items: stretch;
+
+	&__itemsWrapper {
+		display: flex;
+		flex-wrap: nowrap;
+		align-content: center;
+		justify-content: center;
+	}
+
+	&.-compact {
+		#{$self}__items.-compact {
+			display: flex !important;
```

Comment:
Have you tried to display correct layout without using `!important`? It should be the last resort.